qui/tray/domains.py

```diff
@@ -454,35 +513,43 @@ def have_running_and_all_are_paused(self):
                         return False
         return found_paused
 
-    def add_domain_item(self, vm, _event, **_kwargs):
+    def add_domain_item(self, _submitter, event, vm, **_kwargs):
         # check if it already exists
+        vm = self.qapp.domains[str(vm)]
         if vm in self.menu_items:
             return
-        domain_item = DomainMenuItem(vm)
-        position = 0
-        for i in self.tray_menu:  # pylint: disable=not-an-iterable
-            if not hasattr(i, 'vm') \
-               or (i.vm is not None and i.vm.name > vm.name):
-                break
-            position += 1
-        self.tray_menu.insert(domain_item, position)
+        domain_item = DomainMenuItem(vm, self)
+        if not event:
+            self.tray_menu.add(domain_item)
+        else:
+            position = 0
+            for i in self.tray_menu:  # pylint: disable=not-an-iterable
+                if not hasattr(i, 'vm') \
+                   or (i.vm is not None and i.vm.name > vm.name):
+                    break
+                position += 1
+            self.tray_menu.insert(domain_item, position)
         self.menu_items[vm] = domain_item
 
-    def property_change(self, vm, *_args, **_kwargs):
+    def property_change(self, vm, event, *_args, **_kwargs):
         if vm not in self.menu_items:
             return
-        self.menu_items[vm].name.update_tooltip(netvm_changed=True)
+        if event.endswith('netvm'):
```

Comment:
This (and similar thing below) may also match other properties with the same suffix. If you want to avoid writing the whole `property-set:netvm`, then at least include `:` here (`:` is guaranteed to not be part of property name itself).

Reply:
ok, although I don't think it's necessary - I only add a handler to property-set:netvm and property-set:label events, not a wildcard property-set:* handler.